File: Sources/relayout/Log.swift
import Foundation

/// Minimal append-only log at ~/Library/Logs/relayout.log, for diagnosing
/// permission and hotkey issues that have no visible error surface.
enum Log {
    private static let url = FileManager.default
        .homeDirectoryForCurrentUser
        .appendingPathComponent("Library/Logs/relayout.log")

    private static let formatter: DateFormatter = {
        let f = DateFormatter()
        f.dateFormat = "yyyy-MM-dd HH:mm:ss"
        return f
    }()

    static func write(_ message: String) {
        let line = "\(formatter.string(from: Date())) \(message)\n"
        if let handle = try? FileHandle(forWritingTo: url) {
            defer { handle.closeFile() }
            handle.seekToEndOfFile()
            handle.write(Data(line.utf8))
        } else {
            try? Data(line.utf8).write(to: url)
        }
    }
}

File: windows/Log.cs
namespace Relayout;

/// Minimal append-only log at %LOCALAPPDATA%\relayout.log, for diagnosing
/// hotkey and conversion issues that have no visible error surface.
static class Log
{
    static readonly string Path = System.IO.Path.Combine(
        Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
        "relayout.log");

    public static void Write(string message)
    {
        try
        {
            File.AppendAllText(Path, $"{DateTime.Now:yyyy-MM-dd HH:mm:ss} {message}{Environment.NewLine}");
        }
        catch
        {
            // Logging must never take the app down.
        }
    }
}
